src/lib.rs

```diff
@@ -12,6 +12,69 @@ extern crate ethereum_types;
 #[cfg(test)]
 extern crate serde_json;
 
-mod raw_transaction;
+use secp256k1::{SecretKey, Message, Secp256k1};
+use rlp::RlpStream;
 
+mod raw_transaction;
 pub use self::raw_transaction::RawTransaction;
+
+pub trait Transaction {
```

Comment:
Maybe we should rename this trait something else like `EIP2718`, or `TypedTransactionEnvelope`, or some combination of the two? My reasoning being `LegacyTransaction` is separate from typed transactions, so it will not implement this trait. Which seems a bit weird since it is still a transaction.

Reply:
If I rename it, then `TypedTransactionEnvelope` is fitting. I think EIP2718 isn't pleasant as a type name.

However I intended to make the legacy transaction (RawTransaction) implement the new transaction trait. Are you sure the legacy transaction should not implement the trait? A legacy transaction is still a transaction, and can be signed, serialized, hashed, and so on

Reply:
I agree with you on the last part, that the legacy txn can still be signed, serialized, etc. But looking at the spec, it seems that LegacyTransactions are not a part of EIP-2718, ie, they are not typed transactions. The spec says that transactions are either `TransactionType || TransactionPayload` or `LegacyTransaction`. The problem I foresaw was that legacy txn doesn't have a `TransactionType` I don't think, which is a required method of the trait. But I suppose we could just return None. What are your thoughts?